gef-ui/server/controllers/amendments/do-you-have-a-facility-end-date/do-you-have-a-facility-end-date.test.ts

```diff
@@ -0,0 +1,158 @@
+    getFacilityMock.mockResolvedValue({ details: mockFacility });
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should call getApplication with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getDoYouHaveAFacilityEndDate(req, res);
+
+    // Assert
+    expect(getApplicationMock).toHaveBeenCalledTimes(1);
+    expect(getApplicationMock).toHaveBeenCalledWith({ dealId, userToken: req.session.userToken });
+  });
+
+  it('should call getFacility with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getDoYouHaveAFacilityEndDate(req, res);
+
+    // Assert
+    expect(getFacilityMock).toHaveBeenCalledTimes(1);
+    expect(getFacilityMock).toHaveBeenCalledWith({ facilityId, userToken: req.session.userToken });
+  });
+
+  it('should render the facility value template if the facility is valid', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getDoYouHaveAFacilityEndDate(req, res);
+
+    // Assert
+    const expectedRenderData: DoYouHaveAFacilityEndDateViewModel = {
+      exporterName: companyName,
+      cancelUrl: `/gef/application-details/${dealId}/facility/${facilityId}/amendments/${amendmentId}/cancel`,
+      previousPage: `/gef/application-details/${dealId}/facility/${facilityId}/amendments/${amendmentId}/new-cover-end-date`,
+    };
+
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+    expect(res._getRenderView()).toEqual('partials/amendments/do-you-have-a-facility-end-date.njk');
+    expect(res._getRenderData()).toEqual(expectedRenderData);
+  });
+
+  it('should redirect if the facility is not found', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+    getFacilityMock.mockResolvedValue({ details: undefined });
+
+    // Act
+    await getDoYouHaveAFacilityEndDate(req, res);
+
+    // Assert
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.Found);
+  });
+
+  it('should redirect if the deal is not found', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+    getApplicationMock.mockResolvedValue(undefined);
+
+    // Act
+    await getDoYouHaveAFacilityEndDate(req, res);
+
+    // Assert
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.Found);
+  });
+
+  it('should redirect if the facility cannot be amended', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+    getFacilityMock.mockResolvedValue({ details: { ...mockFacility, hasBeenIssued: false } });
+
+    // Act
+    await getDoYouHaveAFacilityEndDate(req, res);
+
+    // Assert
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.Found);
+  });
+
+  it('should render `problem with service` if getApplication throws an error', async () => {
+    // Arrange
+    getApplicationMock.mockRejectedValue(new Error('test error'));
```

Comment:
to be more dry, could declare new Error('test error') on the top

Reply:
Whoever picks this up, we should update other amendment tests to do the same, to be consistent.